proposals/46-store-my-text-example-app.md

```diff
@@ -0,0 +1,111 @@
+# Store-my-text Filecoin example app
+
+Authors: @zenground0 @rvagg
+
+Initial PR: https://github.com/protocol/web3-dev-team/pull/46
+
+## Purpose &amp; impact 
+
+#### Background &amp; intent
+
+_Describe the desired state of the world after this project? Why does that matter?_
+
+We have an extremely simple example application, _and_ a functioning service that has real-world utility that exercises Filecoin and deal-making.
+
+ * A **very** simple web application, minimal UI fluff and distraction: **Save my text to Filecoin**
+   - Text field to enter some UTF-8 text
+   - Field to submit your email address
+   - A button to "Submit"
+ * User enters some non-empty text and their email address and presses "Submit"
+ * Application performs a simple confirmation-of-email ("click this link") to confirm a real user and manage trivial DoS
+ * Application handles storage of text on Filecoin and sends confirmation email
+   - Application mediates the storage deal-making process for small content - finding miners, batching content, etc.
+ * CID provides persistent "proof-of-text" to the user
+
+Initially this is just a very simple system that demonstrates (and exercises, and educates us on) the full deal flow for simple content.
+
+There's plenty of extensions that could be made to this, turning it into a genuinely useful application, such as:
+
+ * IPFS integration
+ * Other input types (images, etc. getting toward the NFT use-case)
+ * Payments ("Deposit X FIL to this address...", "Enter your CC to purchase X FIL")
+ * ETH integration to turn content into an NFT (ERC721, ERC1155, whatever is needed, maybe integration with an existing NFT service?)
+
+#### Assumptions &amp; hypotheses
+
+_What must be true for this project to matter?_
+
+ * when attempting to do deal, frequently it is difficult to get it completed
+ * Building applications that perform simple & small storage deals to Filecoin is currently difficult
+ * Web3 developers want solid but grokkable example applications they can copy (fork-and-go!)
+
+#### User workflow example
+
+_How would a developer or user use this new capability?_
+
+ * Play with application live (we host and manage and pay for storage of small amounts of text, with limits per email address)
+ * Fork open source application
+ * Make new web3 app with Filecoin storage
+ * Profit
+
+#### Impact
+
+_How would this directly contribute to web3 dev stack product-market fit?_
+
+ * Exercise deal-making for small resources
+ * Provide a very simple example for web3 app developers to build on
+
```

Comment:
1.  Team learning is legitimate and necessary; it contributes to PMF because we have to understand the developer journey (esp for new devs) before we can improve it.

2.  Product (me) can learn and use those findings to better identify the top problems facing new developers.